### PR TITLE
NO-37-JoyDoc-Mobile-Renderer

### PR DESCRIPTION
**Pull request by swutask**
- JoyDoc Mobile Renderer

Loom Video:
https://www.loom.com/share/cba3e4feaf134a29b3d8db9c08ce5024?sid=7fb7fcc0-4cc3-40d0-af2a-d1084811d9c4

### CODE REVIEW

File: Sources/joyfill-components-swift/Components/ComponentViewController/APIComponents/API/JoyfillAPI.swift [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+                DispatchQueue.main.async {
+                    componentTableView.reloadData()
+                }
+            }
+            if componentTypeValue == "text" {
+                textFieldString = string
+            }
+            
+        case .valueElementArray(let valueElements):
+            if componentTypeValue == "chart" {
```
Comment (jeremypagley) [inline Sources/joyfill-components-swift/Components/ComponentViewController/APIComponents/API/JoyfillAPI.swift:277]: Looks like this is happening in multiple places here. What is the difference between this one and the one above?

We should not be doing data mapping in multiple places.

File: Sources/joyfill-components-swift/Components/ComponentViewController/APIComponents/ComponentViewController.swift [hunk 2]
```diff
@@ -109,8 +103,8 @@ extension ComponentsViewController: UITableViewDelegate, UITableViewDataSource {
                 
                 // MARK: MultiChoice Function Call From Package
                 let multipleChoice = MultipleChoice()
-                tableView.rowHeight = 200
-                multipleChoice.titleFontSize = 20
+                tableView.rowHeight = 190
+                multipleChoice.multiSelect = true
```
Comment (jeremypagley) [inline Sources/joyfill-components-swift/Components/ComponentViewController/APIComponents/ComponentViewController.swift:107]: Why are there pre-set values here in the renderer? Wouldn't things like this be in the components and then options for this field would be passed in.

File: Sources/joyfill-components-swift/Components/ComponentViewController/APIComponents/ComponentViewController.swift [hunk 3]
```diff
@@ -181,7 +169,7 @@ extension ComponentsViewController: UITableViewDelegate, UITableViewDataSource {
                 tableView.rowHeight = 100
                 number.numberFieldPlacholder = "0"
                 number.titleText = componentHeaderText[i]
-                number.currentPage = Int(numberFieldString ?? 0)
+                number.currentPage = numberFieldString ?? 0
                 number.numberField.text = "\(numberFieldString ?? 0)"
                 number.frame = CGRect(x: 10, y: 10, width: tableView.bounds.width - 20, height: tableView.rowHeight)
```
Comment (jeremypagley) [inline Sources/joyfill-components-swift/Components/ComponentViewController/APIComponents/ComponentViewController.swift:174]: Wouldn't you want things like the rect setup to be within the component of Signature and not set as your mapping it out?

Ideally you would just detect which field is on this Document, then map out each one with one line of code instantiating the component and then passing data into them. It looks like you are grabbing an instance of each one and creating its UI on the fly?

I may be confused on what this is doing so let me know

File: Sources/joyfill-components-swift/Components/ComponentViewController/JSONComponents/ComponentsViewController.swift [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+            componentTableView.leadingAnchor.constraint(equalTo: componentView.leadingAnchor, constant: 0),
+            componentTableView.trailingAnchor.constraint(equalTo: componentView.trailingAnchor, constant: 0),
+            componentTableView.bottomAnchor.constraint(equalTo: componentView.bottomAnchor, constant: -10)
+        ])
+        
+        // Set cornerRadius and shadow to view.
+        backgroundColor = .white
+        componentView.backgroundColor = .white
+        componentView.layer.cornerRadius = 15
+        componentView.layer.shadowOpacity = 0.3
```
Comment (jeremypagley) [inline Sources/joyfill-components-swift/Components/ComponentViewController/JSONComponents/ComponentsViewController.swift:61]: What does the componentView reference

**Review by jeremypagley — CHANGES_REQUESTED** [review]

File: Sources/joyfill-components-swift/Components/ComponentViewController/APIComponents/API/JoyfillAPI.swift (hunk 1 above)
Comment (swutask) [reply Sources/joyfill-components-swift/Components/ComponentViewController/APIComponents/API/JoyfillAPI.swift:277]: My apologize for this as I was trying something else based on the functionality like to keep the separate code mobile view and primary view. You're right and I've changes it now

File: Sources/joyfill-components-swift/Components/ComponentViewController/APIComponents/ComponentViewController.swift (hunk 2 above)
Comment (swutask) [reply Sources/joyfill-components-swift/Components/ComponentViewController/APIComponents/ComponentViewController.swift:107]: It was missed on my side. I've corrected with the api

File: Sources/joyfill-components-swift/Components/ComponentViewController/JSONComponents/ComponentsViewController.swift (hunk 4 above)
Comment (swutask) [reply Sources/joyfill-components-swift/Components/ComponentViewController/JSONComponents/ComponentsViewController.swift:61]: It's the view on which all the components will render

File: Sources/joyfill-components-swift/Components/ComponentViewController/APIComponents/ComponentViewController.swift (hunk 3 above)
Comment (swutask) [reply Sources/joyfill-components-swift/Components/ComponentViewController/APIComponents/ComponentViewController.swift:174]: I'm doing exactly the same as you thought like to detect which component  is on the document and then map each one with the component and passing the value/data to each component coming from the API

**Pull request merged**